### PR TITLE
[Feature] Subtask - UploadTaskBlock

### PR DESCRIPTION
**Pull request by MatthewK3023**
this PR is to fix the subtask "Upload Job need to block Setup Job"
https://www.notion.so/51c73a064a244f7dbcb4abeb12a268f7?v=3d8bfc9cbca44349924fc533dc6d4d61&p=04962dc793d147e58d90ec8470aaaa8e

### CODE REVIEW

**Review by NoctisHong — CHANGES_REQUESTED** [review]
Review data_setup

File: src/decanter/core/jobs/data_setup.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# pylint: disable=C0103
+# pylint: disable=too-many-instance-attributes
+"""DataUpload
+
+Handle data upload to Decanter Core server Stores Data
+attribute.
```
[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py:6]: done

File: src/decanter/core/jobs/data_setup.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+
+class DataSetup(Job):
+    """DataUpload manage to get the result from data upload.
+
+    Handle the execution of upload task in order to upload data to
+    Decanter Core server Stores the upload results in DataUpload attributes.
```
[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py]: done

[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py]: done

File: src/decanter/core/jobs/data_setup.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        originSchema (dict): The original data schema.
+        annotationsMeta (dict): information: Extra information for data.
+        options (dict): Extra information for data.
+        created_at (str): The date the data was created.
+        updated_at (str): The time the data was last updated.
+        completed_at (str): The time the data was completed at.
+        name (str): Name to track Job progress, will give default name if None.
+    """
+    def __init__(self, train_data=None, setup_params=None, name=None):
+        """DataUpload Init.
```
[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py]: done

File: src/decanter/core/jobs/data_setup.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+from decanter.core.extra.decorators import update
+from decanter.core.extra.utils import check_response, gen_id
+from decanter.core.jobs.job import Job
+from decanter.core.jobs.task import SetupTask
+
+
+logger = logging.getLogger(__name__)
+
+
+class DataSetup(Job):
```
[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py]: done

File: src/decanter/core/jobs/data_setup.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    """
+    def __init__(self, train_data=None, setup_params=None, name=None):
+        """DataUpload Init.
+
+        Args:
+            train_data (:obj:`~decanter.core.jobs.data_upload.DataUpload`)
+            setup_params (:obj:`dict`, optional): Settings for setup data.
+            name (:obj:`str`, optional): Name to track Job progress
+        """
+        super().__init__(jobs=[train_data],
```
[the remark this answers is not in the document]
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py]: done

File: src/decanter/core/jobs/experiment.py [hunk 6]
```diff
@@ -48,7 +49,9 @@ class Experiment(Job):
     """
     def __init__(self, train_input, select_model_by='mse', name=None):
         super().__init__(
-            jobs=[train_input.data],
+            jobs=[train_input.data, train_input.data.jobs[0]]
```
Comment (NoctisHong) [inline src/decanter/core/jobs/experiment.py]: After adding the correct dependency between DataSetup and DataUpload jobs,
Experiment job will have the denpendency of corresponded data.
So threre should be no changes hrere.

**Review by NoctisHong — APPROVED** [review]

**Pull request merged**